backend/src/main/java/com/woowacourse/ternoko/common/JwtProvider.java

```diff
@@ -0,0 +1,63 @@
+package com.woowacourse.ternoko.common;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Date;
+
+import javax.crypto.SecretKey;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import com.woowacourse.ternoko.common.exception.ExceptionType;
+import com.woowacourse.ternoko.common.exception.ExpiredTokenException;
+import com.woowacourse.ternoko.common.exception.InvalidTokenException;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.ExpiredJwtException;
+import io.jsonwebtoken.Jws;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.security.Keys;
+
+@Component
+public class JwtProvider {
+
+	private final SecretKey key;
+	private final long validityInMilliseconds;
+
+	public JwtProvider(@Value("${security.jwt.token.secret-key}") final String secretKey,
+		@Value("${security.jwt.token.expire-length}") final long validityInMilliseconds) {
+		this.key = Keys.hmacShaKeyFor(secretKey.getBytes(StandardCharsets.UTF_8));
+		this.validityInMilliseconds = validityInMilliseconds;
+	}
+
+	public String createToken(final String payload) {
+		Date now = new Date();
+		Date validity = new Date(now.getTime() + validityInMilliseconds);
+		return Jwts.builder()
+			.setSubject(payload)
+			.setIssuedAt(now)
+			.setExpiration(validity)
+			.signWith(key, SignatureAlgorithm.HS256)
+			.compact();
+	}
+
+	public String getPayload(final String token) {
+		return Jwts.parserBuilder()
+			.setSigningKey(key).build()
+			.parseClaimsJws(token)
+			.getBody()
+			.getSubject();
+	}
+
+	public void validateToken(final String token) {
+		try {
+			Jws<Claims> claims = Jwts.parser().setSigningKey(key).parseClaimsJws(token);
+			claims.getBody().getExpiration().before(new Date());
+		} catch (ExpiredJwtException e) {
+			throw new ExpiredTokenException(ExceptionType.EXPIRED_TOKEN);
```

Comment:
`claims.getBody().getExpiration().before(new Date());` 코드의 결과가 무시되고 있는 것 같아요!
혹시, 토큰의 만료 기간을 검증하고 예외를 발생하는 로직이었을까요?

그렇다면
```suggestion
		try {
            final Jws<Claims> claims = Jwts.parser().setSigningKey(key).parseClaimsJws(token);
            final boolean expired = claims.getBody().getExpiration().after(new Date());
            if (expired) {
                throw new ExpiredTokenException(ExceptionType.EXPIRED_TOKEN);
            }
```

다음과 같은 방법에 대해선 어떻게 생각하시나요?

Reply:
claims 내에 있는 메서드를 처음 써봐서, 제대로 파악을 하지 못하고 작성했던 것 같아요. 🥲 덕분에 해당 메서드에 대한 이해도가 높아져서 아주 좋네요 👍🏻

Reply:
다시 확인해보니까 parseClaimsJws 하는 과정에서 메서드를 호출하는 시점에 클레임이 만료된 경우 
`ExpiredJwtException` 을 던져 주네요.  위 로직 자체가 없어도 되겠어요. 


> <img width="635" alt="스크린샷 2022-07-26 오후 4 22 59" src="https://user-images.githubusercontent.com/26570275/180947942-380ae4dc-057a-484a-8f50-484a90cbfedc.png">
,